packages/protocol-kit/src/Safe.ts

```diff
@@ -214,13 +222,32 @@ class Safe {
     return this.#contractManager.multiSendCallOnlyContract.getAddress()
   }
 
+  /**
+   * Checks if the current Safe is deployed.
+   *
+   * @returns TRUE if the Safe contract is deployed
+   */
+  async isSafeDeployed(): Promise<boolean> {
+    const safeAddress = await this.getAddress()
+    const isSafeDeployed = await this.#ethAdapter.isContractDeployed(safeAddress)
+    return isSafeDeployed
+  }
+
   /**
    * Returns the Safe Master Copy contract version.
    *
    * @returns The Safe Master Copy contract version
    */
   async getContractVersion(): Promise<SafeVersion> {
-    return this.#contractManager.safeContract.getVersion()
+    if (this.#contractManager.safeContract) {
+      return this.#contractManager.safeContract.getVersion()
+    }
+
+    if (this.#predictedSafe?.safeDeploymentConfig.safeVersion) {
+      return this.#predictedSafe.safeDeploymentConfig.safeVersion
```

Comment:
Since the function is async, it's already will be treated as a resolved promise. You'd only need `promise.resolve` only if the function wasn't async but should return a promise

Reply:
ts playground: https://www.typescriptlang.org/play?target=2#code/IYZwngdgxgBAZgV2gFwJYHsI2QUxMgRgAoBKALhgAUAndAW1RBwB59rUIBzAPhgG8Y1HMgTUsAcnzA0sNh07iYAXwDcAKDWIUGLLnykKNeoxZyuvAUJFiYk5NNSzk7LotVA